client/handlers.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/mikloslorinczi/infra-exec/common"
+	"github.com/mikloslorinczi/infra-exec/executor"
+	"github.com/pkg/errors"
+)
+
+// CompareTags checks if all the tags of the Task matches with the Node's tags.
+func compareTags(taskTags, nodeTags string) bool {
+	taskSplit := strings.Split(taskTags, " ")
+	nodeSplit := strings.Split(nodeTags, " ")
+	for _, taskTag := range taskSplit {
+		found := false
+		for _, nodeTag := range nodeSplit {
+			if taskTag == nodeTag {
+				found = true
+			}
+		}
+		if !found {
```

Comment:
`return found` ?
What is the reason of having an extra `if` block here?

Reply:
Should be easier to apply a short cut logic here? e.g.:
```
        taskSplit := strings.Split(taskTags, " ")
	nodeSplit := strings.Split(nodeTags, " ")
         # slices above would be sorted alphabetically as well 
         # and transformed to either lower or upper case for a case insensitive use-case
	for _, taskTag := range taskSplit {
		for _, nodeTag := range nodeSplit {
			if taskTag == nodeTag {
				return true
			}
		}
	}
	return false
```

Reply:
If I understand correctly all of the Task's tags needs to be matched, not just the first.